gneiss/regression/_regression.py

```diff
@@ -5,86 +5,54 @@
-
-    Returns
-    -------
-    pd.DataFrame
-        Subset of `table` with common row names as `metadata`
-        and common columns as `tree.tips()`
-    pd.DataFrame
-        Subset of `metadata` with common row names as `table`
-    skbio.TreeNode
-        Subtree of `tree` with common tips as `table`
-    """
-    if np.any(table <= 0):
-        raise ValueError('Cannot handle zeros or negative values in `table`. '
-                         'Use pseudocounts or ``multiplicative_replacement``.'
-                         )
-    # check to see if there are overlapping nodes in tree and table
-    overlap = {n.name for n in tree.traverse()} & set(table.columns)
-    if len(overlap) == 0:
-        raise ValueError('There are no internal nodes in `tree` after'
-                         'intersection with `table`.')
-
-    _table, _metadata = match(table, metadata)
-    _table, _tree = match_tips(_table, tree)
-    non_tips_no_name = [(n.name is None) for n in _tree.levelorder()
-                        if not n.is_tip()]
-
-    if any(non_tips_no_name):
-        _tree = rename_internal_nodes(_tree)
-    return _table, _metadata, _tree
-
-
-def _to_balances(table, tree):
-    """ Converts a table of abundances to balances given a tree.
-
-    Parameters
-    ----------
-    table : pd.DataFrame
-        Contingency table where samples correspond to rows and
-        features correspond to columns.
-    tree : skbio.TreeNode
-        Tree object where the leaves correspond to the columns contained in
-        the table.
-
-    Returns
-    -------
-    pd.DataFrame
-        Contingency table where samples correspond to rows and
-        balances correspond to columns.
-    np.array
-        Orthonormal basis in the Aitchison simplex generated from `tree`.
-    """
-    non_tips = [n.name for n in tree.levelorder() if not n.is_tip()]
-    basis, _ = balance_basis(tree)
-
-    mat = ilr(table.values, basis=basis)
-    ilr_table = pd.DataFrame(mat,
-                             columns=non_tips,
-                             index=table.index)
-    return ilr_table, basis
+import skbio
+from ._ols import OLSModel, ols
+from ._mixedlm import LMEModel, mixedlm
+
+from q2_composition.plugin_setup import Composition
+from q2_types.feature_table import FeatureTable
+from q2_types.tree import Phylogeny, Rooted, Unrooted
+from qiime2.plugin import Str, Metadata
+from gneiss.plugin_setup import plugin
+from ._type import Regression_g, Linear_g, LinearMixedEffects_g
+
+
+def ols_regression(table: pd.DataFrame, tree: skbio.TreeNode,
+                   metadata: Metadata, formula: str) -> OLSModel:
+    res = ols(table=table, tree=tree, metadata=metadata._dataframe,
+              formula=formula)
+    res.fit()
+    return res
+
+
+plugin.methods.register_function(
+    function=ols_regression,
+    inputs={'table': FeatureTable[Composition],
+            'tree': Phylogeny[Rooted | Unrooted]},
+    parameters={'formula': Str, 'metadata': Metadata},
```

Comment:
Can you add parameter and output descriptions? For example, see [here](https://github.com/qiime2/q2-diversity/blob/master/q2_diversity/plugin_setup.py#L35-L47).